### ISSUE TITLE
Medaka exception error

### ISSUE THREAD

**Issue by khyojoo01** [nanoporetech/medaka#97]
I tried to run medaka_consensus on human genome generated by ONT nanopore using following command
```
medaka_consensus -i inputfile.fastq -d draft-made-using-wtdbg2.ctg.fa -o $outputdirectory/medaka_consensus -t 8 -m r941_prom_fast
```
and got exception error message and it stopped. actually job didn't stop, it froze.

```
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/threading.py", line 923, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5/threading.py", line 871, in run
    self._target(*self._args, **self._kwargs)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/prediction.py", line 251, in _fill_parallel
    samples, remain = fut.result()
  File "/usr/local/lib/python3.5/concurrent/futures/_base.py", line 398, in result
    return self.__get_result()
  File "/usr/local/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/prediction.py", line 287, in _run_region
    return data_gen.samples, data_gen._quarantined
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/features.py", line 612, in samples
    all_data.extend(chunks)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/common.py", line 310, in chunks
    for i, pos in enumerate(chunkers['positions']):
KeyError: 'positions'
```
it came after these messages
```
[20:01:18 - Predict] Setting tensorflow threads to 8.
[20:01:18 - Predict] Processing 5541 long region(s) with batching.
[20:01:18 - Predict] Using model: /data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/data/r941_prom_fast_model.hdf5.
[20:01:18 - ModelLoad] Building model with cudnn optimization: False
[20:01:20 - DLoader] Initializing data loader
[20:01:20 - PWorker] Running inference for 2626.5M draft bases.
[20:01:20 - Sampler] Initializing sampler for consensus of region ctg554:0-1000000.
[20:01:20 - Sampler] Initializing sampler for consensus of region ctg554:0-1450375.
[20:01:20 - Sampler] Initializing sampler for consensus of region ctg3245:0-12517.
[20:01:20 - Sampler] Initializing sampler for consensus of region ctg1457:0-95251.
[20:01:22 - Feature] Processed ctg1457:0.0-95250.0 (median depth 21.0)
[20:01:22 - Sampler] Took 2.13s to make features.
[20:01:22 - Feature] Pileup counts do not span requested region, requested ctg554:0-1450375, received 999000-1450365.
[20:01:22 - Feature] Pileup counts do not span requested region, requested ctg3245:0-12517, received 2-12516.
[20:01:22 - Feature] Processed ctg3245:2.0-12516.0 (median depth 294.0)
[20:01:22 - Sampler] Took 2.17s to make features.
[20:01:22 - Feature] Processed ctg554:999000.0-1450365.0 (median depth 16.0)
[20:01:22 - Sampler] Took 2.30s to make features.
[20:01:22 - Feature] Processed ctg554:0.0-999999.0 (median depth 20.0)
[20:01:22 - Sampler] Took 2.62s to make features.
```
and there were outputs like this
```
80265148845 Oct 24 19:44 calls_to_draft.bam
20260968 Oct 24 20:00 calls_to_draft.bam.bai
5896 Oct 24 20:01 consensus_probs.hdf
```

Please help me.

**Environment (if you do not have a GPU, write No GPU):**
 - OS: CentOS release 6.10 (Final)

**Comment by cjw85** [comment]
Hi @khyojoo01,

I've been looking at the code to understand how this would happen. If possible it would be useful to have your data, is it something you can share?

I see that your file is quite large. We might be able to narrow down the data causing the problem by processing the contigs individually. From your output above I would guess that one of the contigs `ctg1457, ctg3245, ctg554` is at fault. Could you try running

    for ctg in ctg1457 ctg3245 ctg554; do
        medaka consensus calls_to_draft.bam $ctg.hdf --regions $ctg
    done

to determine which one is causing medaka to throw the exception?

**Comment by cjw85** [comment]
@khyojoo01 

Could you report the version of python that you are using, the output of:

    python --version

we suspect this may be the issue (https://bugs.python.org/issue24931).

**Comment by khyojoo01** [comment]
Well, my python version is Python 2.7.15
I'm so sorry but I'm afraid I can't share my data with you. But I'm running suspected contig testing code so I'll let you know what happened after that.
Thank you for your help.

**Comment by khyojoo01** [comment]
While I was writing that comment, I got result of it and I'm posting it here right now.
```
[14:34:55 - Predict] Processing region(s): ctg1457:0-95251
[14:34:56 - Predict] Setting tensorflow threads to 1.
[14:34:57 - Predict] Processing 1 long region(s) with batching.
[14:34:57 - Predict] Using model: /data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/data/r941_min_high_model.hdf5.
[14:34:57 - ModelLoad] Building model with cudnn optimization: False
[14:34:59 - DLoader] Initializing data loader
[14:34:59 - Sampler] Initializing sampler for consensus of region ctg1457:0-95251.
[14:34:59 - PWorker] Running inference for 0.1M draft bases.
[14:35:06 - Feature] Processed ctg1457:0.0-95250.0 (median depth 21.0)
[14:35:06 - Sampler] Took 6.47s to make features.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/threading.py", line 923, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5/threading.py", line 871, in run
    self._target(*self._args, **self._kwargs)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/prediction.py", line 276, in _fill_parallel
    samples, remain = fut.result()
  File "/usr/local/lib/python3.5/concurrent/futures/_base.py", line 398, in result
    return self.__get_result()
  File "/usr/local/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/prediction.py", line 287, in _run_region
    return data_gen.samples, data_gen._quarantined
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/features.py", line 612, in samples
    all_data.extend(chunks)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/common.py", line 310, in chunks
    for i, pos in enumerate(chunkers['positions']):
KeyError: 'positions'
```

**Comment by khyojoo01** [comment]
And after that moment, I tried Keyboard interruption with ^C and got these.
```
KeyboardInterrupt
^CTraceback (most recent call last):
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/bin/medaka", line 6, in <module>
    from medaka.medaka import main
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/medaka.py", line 4, in <module>
    from pkg_resources import resource_filename
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/pkg_resources/__init__.py", line 3250, in <module>
    @_call_aside
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/pkg_resources/__init__.py", line 3234, in _call_aside
    f(*args, **kwargs)
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/pkg_resources/__init__.py", line 3286, in _initialize_master_working_set
    list(map(working_set.add_entry, sys.path))
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/pkg_resources/__init__.py", line 623, in add_entry
    for dist in find_distributions(entry, True):
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2065, in find_on_path
    for dist in factory(fullpath):
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2127, in distributions_from_metadata
    if len(os.listdir(path)) == 0:
KeyboardInterrupt
^CTraceback (most recent call last):
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/bin/medaka", line 6, in <module>
    from medaka.medaka import main
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/medaka.py", line 10, in <module>
    import medaka.datastore
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/medaka/datastore.py", line 14, in <module>
    import h5py
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/h5py/__init__.py", line 62, in <module>
    from .tests import run_tests
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/h5py/tests/__init__.py", line 15, in <module>
    from . import old, hl
  File "/data2/2013161046/Nanopore_WGS/0_RawData/medaka-PMAU-0011-T-A1-chr21-22-191005/medaka/lib/python3.5/site-packages/h5py/tests/old/__init__.py", line 4, in <module>
    from . import ( test_attrs,
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 330, in __exit__
  File "<frozen importlib._bootstrap>", line 227, in _verbose_message
KeyboardInterrupt
(medaka) [2013161046@master medaka_consensus]$ ll
total 78403992
-rw-rw-r-- 1 2013161046 2013161046 80265148845 Oct 25 04:44 calls_to_draft.bam
-rw-rw-r-- 1 2013161046 2013161046    20260968 Oct 25 05:00 calls_to_draft.bam.bai
-rw-rw-r-- 1 2013161046 2013161046        5896 Oct 25 05:01 consensus_probs.hdf
-rwxrwxr-x 1 2013161046 2013161046         107 Oct 29 14:31 contigtest.sh
-rw-rw-r-- 1 2013161046 2013161046        5896 Oct 29 14:37 ctg1457.hdf

```

**Comment by cjw85** [comment]
Your errors indicate that you are using a flavour of python 3.5 (see the references to `/usr/local/lib/python3.5`), not python 2.7.15. More precisely we suspect you are using python 3.5.0 which will cause the behaviour you are seeing.

You will need to upgrade the python version on your system, or probably more simply use conda to install medaka by following the instructions in the README.

**Comment by khyojoo01** [comment]
ok I'll try them and let you know if needed. Thank you for your comments.

**Comment by cjw85** [comment]
We have put a patch into a forthcoming release that will correct this error for python 3.5.0.

**Comment by cjw85** [comment]
We have just released v0.10.1 which will appear on pypi shortly. It contains a workaround for the bug in python 3.5.0.
